pulp_container/app/authorization.py

```diff
@@ -133,6 +133,13 @@ def determine_access(self):
         actions = set(actions.split(","))
 
         permitted_actions = set()
+        if "push" in actions:
+            has_permission = self.actions_permissions["push"](name)
+            if has_permission:
+                permitted_actions.add("push")
+                permitted_actions.add("pull")
+
+            return {"type": typ, "name": name, "actions": list(permitted_actions)}
```

Comment:
What happens if there is push and pull requested (aka in `actions`) but `has_permission` is `False` after checking the push perm?
It looks to me that we won't give even pull perms in such case, is it intentional? Or maybe I'm missing something.

Reply:
no, your observations are right, i need to rework logic a bit, ty

Reply:
It might be worth adding a test which covers that case.